scripts/code_coverage_report/generate_code_coverage_report/Sources/IncrementalCoverageReportGenerator/main.swift

```diff
@@ -0,0 +1,199 @@
+    '[{"file": "FirebaseDatabase/Sources/Api/FIRDataSnapshot.m", "added_lines": [105,106,107,108,109]},{"file": "FirebaseDatabase/Sources/Core/Utilities/FPath.m", "added_lines": [304,305,306,307,308,309]}]'
+    """,
+    transform: FileIncrementalChanges.dataModel
+  )
+  var changedFiles: [FileIncrementalChanges]
+
+  @Option(
+    help: "Uncovered line JSON file output path"
+  )
+  var uncovered_line_file_json: String = Constants.DEFAULT_UNCOVERED_LINE_REPORT_FILE_NAME
+
+  func readFile(_ fileURL: URL) -> String {
+    var fileContent = ""
+    do {
+      fileContent = try String(contentsOf: fileURL)
+    } catch {
+      print("Failed reading from URL: \(fileURL), Error: " + error.localizedDescription)
+    }
+    return fileContent
+  }
+
+  // This will transfer line executions report from a xcresult bundle to an array of LineCoverage.
+  // The output will have the file name, line execution counts and the xcresult bundle name.
+  func createLineCoverageRecord(from coverageFile: String, changedFile: String,
+                                lineCoverage: [LineCoverage]? = nil) -> [LineCoverage] {
+    // The indices of the array represent the (line index - 1), while the value is the execution counts.
+    var lineExecutionCounts: [Int?] = []
+    if let dir = URL(string: "file://" + FileManager.default.currentDirectoryPath) {
+      let fileURL = dir.appendingPathComponent(coverageFile)
+      let inString = readFile(fileURL)
+      let lineCoverageRegex = try! NSRegularExpression(pattern: Constants
+        .LINE_EXECUTION_COUNT_PATTERN)
+      for line in inString.components(separatedBy: "\n") {
+        let range = NSRange(location: 0, length: line.utf16.count)
+        if let match = lineCoverageRegex.firstMatch(in: line, options: [], range: range) {
+          // Get the execution counts and append. Line indices are
+          // consecutive and so the array will have counts for each line
+          // and nil for unexecutable lines, e.g. comments.
+          let nsRange = match.range(at: 2)
+          if let range = Range(nsRange, in: line) {
+            lineExecutionCounts.append(Int(line[range]))
+          }
+        }
+      }
+    }
+    if var coverageData = lineCoverage {
+      coverageData
+        .append(LineCoverage(fileName: changedFile, coverage: lineExecutionCounts,
+                             xcresultBundle: coverageFile))
+      return coverageData
+    } else {
+      return [LineCoverage(fileName: changedFile, coverage: lineExecutionCounts,
+                           xcresultBundle: coverageFile)]
+    }
+  }
+
+  // This function is to get union of newly added file lines and lines execution counts, from a xcresult bundle.
+  // Return an array of LineCoverage, which includes uncovered line indices of a file and its xcresult bundle source.
+  func get_uncovered_file_lines(fromDiff changedFiles: [FileIncrementalChanges],
+                                fromXcresult xcresultPath: String,
```

Comment:
Nit: the argument label here should be `xcresultPath` instead of `fromXcresult`,